src/main/java/app/daos/impl/GameDAOImpl.java

```diff
@@ -28,6 +33,34 @@ public Game create(Game game) {
                 throw new EntityExistsException(String.format("Game with id %d already exists", game.getId()));
             }
 
+            if (game.getGenreSet() != null) {
+                Set<Genre> genreSet = new HashSet<>();
+                game.getGenreSet().forEach(genre -> {
+                    Genre foundGenre = em.find(Genre.class, genre.getId());
+
+                    if (foundGenre != null) {
+                        genreSet.add(foundGenre);
+                    } else {
+                        throw new EntityNotFoundException("Genre does not exist.");
```

Comment:
The error message should include the genre id. for example "Genre with id %d could not be found"